pypeit/edgetrace.py

```diff
@@ -4185,16 +4199,29 @@ def maskdesign_matching(self, debug=False):
         # box-slit traces.
         self.clean_traces(rebuild_pca=True)
 
-        # Check that there are traces to match!
+        # Check that there are still traces to match!
         if self.is_empty:
-            msgs.error('No traces to match.')
+            msgs.warn('No edges traced. Slitmask matching cannot be performed')
+            return
 
         # `traceimg` must have knowledge of the flat frame that built it
         self.maskfile = self.traceimg.files[0]
 
         omodel_bspat, omodel_tspat, sortindx, self.slitmask = \
             self.spectrograph.get_maskdef_slitedges(ccdnum=self.traceimg.detector.det, filename=self.maskfile, debug=debug)
 
+        if omodel_bspat[omodel_bspat!=-1].size < 3:
+            msgs.warn('Less than 3 slits are expected on this detector, slitmask matching cannot be performed')
+            # this will allow to catch the boxslit, since in this case slitmask matching is not performed
+            self.par['minimum_slit_gap'] = 0.25
+            self.par['minimum_slit_length_sci'] = 4.5
+            # this will allow to keep the few slits that were found and not have auto_trace fail
+            if not self.can_pca() and self.par['sync_predict'] == 'pca':
+                self.par['sync_predict'] = 'nearest'
```

Comment:
We have generally tried to avoid over-riding `par` within the code (after instantiating the `spectrograph`).

But i can see why this is done here.  
Another approach to consider would be to have the EdgeTrace object hold this parameter.  It would
be initialized to the value from `self.par` but we could
over-ride it as done here.

A bit klunky, but would maintain the "avoid editing par" mantra.

Reply:
I implemented the changes we discussed at the PypeIt meeting